tests/Type/data/collection-generic-static-methods-l948.php

```diff
@@ -44,8 +44,8 @@
 assertType('Illuminate\Support\Collection<App\User, int>', $collection->flip());
 assertType('Illuminate\Support\Collection<int, string>', $items->flip());
 
-assertType('Illuminate\Database\Eloquent\Collection<(int|string), Illuminate\Database\Eloquent\Collection<(int|string), App\User>>', $collection->groupBy('id'));
-assertType('Illuminate\Support\Collection<(int|string), Illuminate\Support\Collection<(int|string), int>>', $items->groupBy('id'));
+assertType('Illuminate\Database\Eloquent\Collection<int, Illuminate\Database\Eloquent\Collection<int, App\User>>', $collection->groupBy('id'));
+assertType("Illuminate\Support\Collection<'', Illuminate\Support\Collection<int, int>>", $items->groupBy('id'));
```

Comment:
If it can't read a property off the types you have in your collection it helpfully groups them all into an empty string group...

Reply:
I get why you do this, although not sure I really care for it...

Reply:
i don't care for it either but it accurately models what laravel is doing with it's types.. i could make it be `new StringType()` but that would give a false sense of security aswell

Reply:
If it can't read the type then why not just default to  `(int|string)` like the previous diff had?

Reply:
because phpstan will report an error when trying to access any other key, alerting the developer to the mistake they have made:

https://phpstan.org/r/45d8567e-0831-4ad5-a35e-fcfe78f2d09c